src/Language/DifferentialDatalog/Syntax.hs

```diff
@@ -230,6 +232,22 @@ instance WithPos ArgType where
 instance PP ArgType where
     pp ArgType{..} = (if atypeMut then "mut" else empty) <+> pp atypeType
 
+data IdentifierWithPos = IdentifierWithPos { idPos :: Pos, idName :: String }
+makeIdentifierWithPos str = IdentifierWithPos nopos str
+
+instance Eq IdentifierWithPos where
+    (==) (IdentifierWithPos _ n1) (IdentifierWithPos _ n2) = n1 == n2
+
+instance Ord IdentifierWithPos where
+    compare (IdentifierWithPos _ n1) (IdentifierWithPos _ n2) = compare n1 n2
+
+instance WithPos IdentifierWithPos where
+    pos = idPos
+    atPos i p = i{idPos = p}
+
+instance PP IdentifierWithPos where
+    pp IdentifierWithPos{..} = pp idName
+
```

Comment:
I think we need `instance PP Name` and use `name` instead of `idName` everywhere.